src/prepared_statement.rs

```diff
@@ -0,0 +1,174 @@
+use byteorder::{BigEndian, WriteBytesExt};
+
+#[derive(Debug)]
+pub struct PreparedStatement {
+    tracing: Option<bool>,
+    warnings: Option<bool>,
+    consistency: Consistency,
+    with_names: bool,
+    page_size: Option<i32>,
+    paging_state: Option<i32>,
+    serial_consistency: Option<Consistency>,
+    timestamp: Option<i64>,
+    frame: Option<Frame>,
+    query_markers: BTreeMap<u32, Value>,
+}
+
+impl Default for PreparedStatement {
+    fn default() -> PreparedStatement {
+        let query_markers = BTreeMap::new();
+
+        PreparedStatement {
+            tracing: None,
+            warnings: None,
+            consistency: Consistency::One,
+            with_names: false,
+            page_size: None,
+            paging_state: None,
+            serial_consistency: None,
+            timestamp: None,
+            frame: None,
+            query_markers: query_markers,
+        }
+    }
+}
+
+impl PreparedStatement {
+    /// `index` represent the position of  marker `?` in the query
+    pub fn set_string(&mut self, index: u32, val: &'static str) -> error::Result<()> {
+        self.query_markers.insert(index, val.to_string().into());
+        Ok(())
+    }
+
+    // /// `index` represent the position of  marker `?` in the query
+    // /// date should be in the format epochtime
+    // not yet working
+    // pub fn set_date(&mut self, index: u32, val: i64) -> error::Result<()> {
+    //     let mut wtr = vec![];
+    //     match wtr.write_i64::<BigEndian>(val) {
+    //         Ok(_) => {
+    //             self.query_markers.insert(index, Value::new_normal(wtr.clone()));
+    //             Ok(())
+    //         }
+    //         Err(err) => return Err(error::Error::from(err)),
+    //     }
+    // }
+
+    /// `index` represent the position of marker `?` in the query
+    pub fn set_int(&mut self, index: u32, val: i32) -> error::Result<()> {
+        let mut wtr = vec![];
+        match wtr.write_i32::<BigEndian>(val) {
+            Ok(_) => {
+                self.query_markers.insert(index, Value::new_normal(wtr.clone()));
+                Ok(())
+            }
+            Err(err) => return Err(error::Error::from(err)),
+        }
+    }
+}
+
+/// This trait once imported into scope would provide the end users to set the values
+/// Prepare and execute statement with prepared Statement
+///
+
+pub trait PrepareAndExecute {
+    /// this returns a prepared statement
+    /// you can call execute_statement after setting the various parameters
+    fn prepare_statement(&mut self,
+                         query: String,
+                         with_tracing: bool,
+                         with_warnings: bool)
+                         -> error::Result<PreparedStatement>;
+
+
+    fn execute_statement(&mut self, prepared_statement: PreparedStatement) -> error::Result<Frame>;
+}
+
+
+impl<'a, T: Authenticator + 'a, X: CDRSTransport + 'a> PrepareAndExecute for Session<T, X> {
+    fn prepare_statement(&mut self,
```

Comment:
Seems to be duplicating https://docs.rs/cdrs/1.0.0-beta.2/cdrs/client/struct.Session.html#method.prepare